### DISCUSSION TITLE
Why is `cid` null, for deletion events?

### DISCUSSION THREAD

**Discussion by DavidBuchanan314** [bluesky-social/atproto#1252] [Clarifications and "Why?"] [answered]
Delete commit operations appear in the firehose with the [repoOp `cid` field](https://github.com/bluesky-social/atproto/blob/7cb8c62b545db8c6139ec83939ff6ac236eca97a/lexicons/com/atproto/sync/subscribeRepos.json#L108) set to null.

https://github.com/bluesky-social/atproto/blob/7cb8c62b545db8c6139ec83939ff6ac236eca97a/packages/pds/src/sequencer/events.ts#L62-L63

IMHO, it would be useful to have it reference the cid of the record that's being deleted.

For "update" operations, it would be nice if there was a way to have them reference both the previous cid, and the new cid.

This would make it easier for clients that are not syncing/storing full repo state (such as feed generators) to verify what's being deleted, and to make sure they're receiving/applying operations in the correct order. (for similar reasons reason that a client might want to pass `swapRecord` when calling `deleteRecord`)

**Comment by bnewbold** [comment, marked as answer]
Here are some middling arguments against including the `cid`:

- extra bytes on the wire, waste of resources in many cases
- draws less attention to the content which has been removed
- implementation performance flexibility

I don't think we have a really strong reason why we don't include it. I can see that it would be helpful in some cases, though I suspect that the majority of downstream services which are trusting the firehose stream (as opposed to keeping a local mirror of the repo to validate events against) wouldn't even be indexing with the CID, just the rkey.
